server/cluster/cluster.go

```diff
@@ -1064,22 +1090,24 @@ func (c *RaftCluster) processRegionHeartbeat(region *core.RegionInfo) error {
 	}
 	region.Inherit(origin, c.GetStoreConfig().IsEnableRegionBucket())
 
-	c.hotStat.CheckWriteAsync(statistics.NewCheckExpiredItemTask(region))
-	c.hotStat.CheckReadAsync(statistics.NewCheckExpiredItemTask(region))
-	reportInterval := region.GetInterval()
-	interval := reportInterval.GetEndTimestamp() - reportInterval.GetStartTimestamp()
-	for _, peer := range region.GetPeers() {
-		peerInfo := core.NewPeerInfo(peer, region.GetWriteLoads(), interval)
-		c.hotStat.CheckWriteAsync(statistics.NewCheckPeerTask(peerInfo, region))
+	if !c.isAPIServiceMode {
+		c.hotStat.CheckWriteAsync(statistics.NewCheckExpiredItemTask(region))
+		c.hotStat.CheckReadAsync(statistics.NewCheckExpiredItemTask(region))
+		reportInterval := region.GetInterval()
+		interval := reportInterval.GetEndTimestamp() - reportInterval.GetStartTimestamp()
+		for _, peer := range region.GetPeers() {
+			peerInfo := core.NewPeerInfo(peer, region.GetWriteLoads(), interval)
+			c.hotStat.CheckWriteAsync(statistics.NewCheckPeerTask(peerInfo, region))
+		}
+		c.coordinator.GetSchedulersController().CheckTransferWitnessLeader(region)
 	}
-	c.coordinator.GetSchedulersController().CheckTransferWitnessLeader(region)
 
 	hasRegionStats := c.regionStats != nil
 	// Save to storage if meta is updated, except for flashback.
 	// Save to cache if meta or leader is updated, or contains any down/pending peer.
 	// Mark isNew if the region in cache does not have leader.
 	isNew, saveKV, saveCache, needSync := regionGuide(region, origin)
-	if !saveKV && !saveCache && !isNew {
+	if !c.isAPIServiceMode && !saveKV && !saveCache && !isNew {
```

Comment:
Shouldn't it be like this?

```suggestion
	if c.isAPIServiceMode || (!saveKV && !saveCache && !isNew) {
```

Reply:
If it is API mode, we should skip it.